x/resource/keeper/msg_server_create_resource.go

```diff
@@ -47,7 +47,9 @@ func (k msgServer) CreateResource(goCtx context.Context, msg *types.MsgCreateRes
 	// Build Resource
 	resource := msg.Payload.ToResource()
 
-	resource.Header.Checksum = sha256.New().Sum(resource.Data)
+	h := sha256.New()
+	h.Write(resource.Data)
+	resource.Header.Checksum = h.Sum(nil)
```

Comment:
Not sure if this logic is correct. Isn't the correct function for this https://pkg.go.dev/crypto/sha256#New?

Reply:
Hey, yeah, that's the correct one.

```go
sha256.Sum256(resource.Data)
```

This outputs a fixed length array type of [32]byte.
So the problem lies in that Protobuf definitions don't allow for fixed byte array lengths, since they are primitively encoded.

I'd suggest creating an addressable (in-memory) array and converting it to a slice that points to those 32 byte values, as below:

```go
checksum := sha256.Sum256([]byte(resource.Data))
resource.Header.Checksum = checksum[:]
```

Reply:
Hey! Both of them are the same and do the same steps inside but I agree It would be more readable and understandable to use just `sha256.Sum256()`